packages/flutter/test/material/tab_bar_theme_test.dart

```diff
@@ -1791,4 +1791,60 @@ void main() {
     targetRect = const Rect.fromLTRB(275.0, 0.0, 325.0, 48.0);
     expectIndicatorAttrs(tabBarBox, rect: rect, targetRect: targetRect);
   });
+
+  testWidgets('splashBorderRadius is passed from TabBarThemeData', (WidgetTester tester) async {
+    const Color hoverColor = Color(0xfff44336);
+    const double radius = 20;
+    await tester.pumpWidget(
+      Material(
+        child: Localizations(
+          locale: const Locale('en', 'US'),
+          delegates: const <LocalizationsDelegate<dynamic>>[
+            DefaultMaterialLocalizations.delegate,
+            DefaultWidgetsLocalizations.delegate,
+          ],
+          child: DefaultTabController(
+            length: 1,
+            child: TabBarTheme(
+              data: TabBarThemeData(
+                 splashBorderRadius: BorderRadius.circular(radius),
+              ),
+              child: TabBar(
+                overlayColor: MaterialStateProperty.resolveWith<Color>(
+                  (Set<MaterialState> states) {
+                    if (states.contains(MaterialState.hovered)) {
+                      return hoverColor;
+                    }
+                    return Colors.black54;
+                  },
+                ),
+
+                tabs: const <Widget>[
```

Comment:
It'd be nice to avoid introducing references to the deprecated `MaterialStateProperty` type, and IMO it'd be great to use a map here to be a bit more concise! :)

```suggestion
              child: const TabBar(
                overlayColor: WidgetStateMapper<Color>(
                  <WidgetStatesConstraint, Color>{
                    WidgetState.hovered: hoverColor,
                    WidgetState.any: Colors.black54,
                  },
                ),

                tabs: <Widget>[
```